test/common.py

```diff
@@ -135,8 +135,21 @@ def assertTensorsSlowEqual(self, x, y, prec=None, message=''):
             max_err = max(max_err, abs(x[index] - y[index]))
         self.assertLessEqual(max_err, prec, message)
 
+    def safeToDense(self, t):
+        r = self.safeCoalesce(t)
+        return r.to_dense()
+
     def safeCoalesce(self, t):
         tc = t.coalesce()
+        self.assertEqual(tc.to_dense(), t.to_dense())
+        self.assertTrue(tc.is_coalesced())
+
+        # Our code below doesn't work when nnz is 0, because
+        # then it's a 0D tensor, not a 2D tensor.
```

Comment:
whaaat?

Reply:
When you zero a sparse tensor, the index and value tensors become zero-dim (old-style, undefined) tensors. This is because it is not possible to have a non zero-dim tensor with zero elements. But you can't transpose a 0D tensor, not in the traditional way anyway.

I wanted to make it so that zeroing a sparse tensor preserved the shapes of indexes and values but apparently this is not easily achieved in the current world order.